File: example/ch06/ex0605.py
class Foo:
    def __init__(self, x):
        self.x = x

def outer():
    f = Foo(10)
    inner1(f)
    print(f.x)  ## 20
    inner2(f)
    print(f.x)  ## これも20
    g = None
    inner2(g)
    print(g is None)  ## True

def inner1(f):
    f.x = 20

def inner2(f):
    f = Foo(30)

outer()

File: example/ch06/ex0605.rb
# coding: utf-8
##
## $ ruby  ex0605.rb

class Foo
  def initialize(x)
    @x = x
  end

  attr_accessor :x

end

def outer
  f = Foo.new(10)
  inner1(f)
  puts f.x
  inner2(f)
  puts f.x
  g = nil
  inner2(g)
  puts g.nil?
end

def inner1(f)
  f.x = 20
end

def inner2(f)
  f = Foo.new(30)
end

outer
